src/entrypoint.sh

```diff
@@ -160,9 +155,20 @@ function git_init() {
     mkdir -p "${HOME}"/.ssh
     ssh-keyscan -t rsa "${source_repo_hostname}" >> "${HOME}"/.ssh/known_hosts
   else
-    info "the source repository is located within GitHub."
-    gh auth setup-git --hostname "${source_repo_hostname}"
-    gh auth status --hostname "${source_repo_hostname}"
+    info "the source repository is located within GitHub."               
+    if [[ -n "${SOURCE_GH_TOKEN}" ]]; then      
+      unset GITHUB_TOKEN      
+      gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${SOURCE_GH_TOKEN}"        
+      if [[ -n "${TARGET_GH_TOKEN}" ]]; then
+        gh auth login --git-protocol "https" --hostname "${SOURCE_REPO_HOSTNAME}" --with-token <<< "${TARGET_GH_TOKEN}"
```

Comment:
what happens if both variables are set?
`SOURCE_GH_TOKEN` and `TARGET_GH_TOKEN`?

Possibly the target needs to be perfomed later (after the pull)